lib/services.js

```diff
@@ -286,9 +286,12 @@ const services = {
      *                           delete a specific version of the object
      * @param {Log} log - logger instance
      * @param {function} cb - callback from async.waterfall in objectGet
+     * @param {boolean} deferDeletion - true if the object should not be removed
+     *                                  from the storage, but be returned
+     *                                  instead.
      * @return {undefined}
      */
-    deleteObject(bucketName, objectMD, objectKey, options, log, cb) {
+    deleteObject(bucketName, objectMD, objectKey, options, log, cb, deferDeletion) {
         log.trace('deleting object from bucket');
```

Comment:
`cb` should be the last param

Reply:
I ended up not changing the function at all, because the azure protection logic is related to the data services, and if we return all locations, we would end up deleting unnecessary things:
* it does not make sense to duplicate the `protectAzureBlocks` call
* yet if we call only a `deleteObjectMD` and `protectAzureBlocks` was supposed to return false, we change the behavior (even if this is harmless, this is not the objective of this PR)

I am sharing here the implementation I did before changing my mind:

```js
    /**
     * Deletes metadata of the object
     * @param {string} bucketName - bucket in which objectMD is stored
     * @param {string} objectKey - object key name
     * @param {object} options - other instructions, such as { versionId } to
     *                           delete a specific version of the object
     * @param {Log} log - logger instance
     * @param {function} cb - callback function
     * @return {undefined}
     */
    deleteMetadata(bucketName, objectKey, options, log, cb) {
        log.trace('Deleting object metadata and returning locations');
        assert.strictEqual(typeof bucketName, 'string');

        return metadata.deleteObjectMD(bucketName, objectKey, options, log,
            (err, res) => {
                if (err) {
                    return cb(err, res);
                }

                log.trace('deleteMetadata: metadata delete OK');
                return cb(err, res);
            });
    },

    /**
     * Deletes objects from a bucket
     * @param {string} bucketName - bucket in which objectMD is stored
     * @param {object} objectMD - object's metadata
     * @param {string} objectKey - object key name
     * @param {object} options - other instructions, such as { versionId } to
     *                           delete a specific version of the object
     * @param {Log} log - logger instance
     * @param {function} cb - callback from async.waterfall in objectGet
     * @return {undefined}
     */
    deleteObject(bucketName, objectMD, objectKey, options, log, cb) {
        log.trace('deleting object from bucket');
        assert.strictEqual(typeof bucketName, 'string');
        assert.strictEqual(typeof objectMD, 'object');
        const objGetInfo = objectMD.location;

        return services.deleteMetadata(bucketName, objectKey, options, log,
            (err, res) => {
                if (err) {
                    return cb(err, res);
                }
                log.trace('deleteObject: metadata delete OK');
                const deleteLog =
                    logger.newRequestLoggerFromSerializedUids(
                        log.getSerializedUids());
                if (objGetInfo === null) {
                    return cb(null, res);
                } else if (!Array.isArray(objGetInfo)) {
                    data.delete(objGetInfo, deleteLog);
                    return cb(null, res);
                }
                return data.batchDelete(objGetInfo, null, null,
                    deleteLog, err => {
                        if (err) {
                            return cb(err);
                        }
                        return cb(null, res);
                    });
            });
    },
```